lib/matplotlib/__init__.py

```diff
@@ -1140,12 +1140,15 @@ def use(backend, *, force=True):
         # Nothing to do if the requested backend is already set
         pass
     else:
-        try:
-            from matplotlib import pyplot as plt
-            plt.switch_backend(name)
-        except ImportError:
-            if force:
-                raise
+        if 'matplotlib.pyplot' in sys.modules:
+            try:
+                from matplotlib import pyplot as plt
+                plt.switch_backend(name)
+            except ImportError:
```

Comment:
Would rather `plt = sys.modules.get("matplotlib.pyplot"); if plt: ...` or at least move the import out of the try... except as that try... except is actually only protecting the call to switch_backend (after this PR), not the pyplot import (which will always succeed) despite what it looks like.

Reply:
This was don, but didn't get auto-folded.